src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/samples/RESTSampleSequenceFilesController.java

```diff
@@ -530,30 +541,38 @@ public ModelMap addNewSequenceFilePairToSample(@PathVariable Long sampleId,
 			logger.trace("Added sequencing run to files" + runId);
 		}
 
-		// add the files and join
-		SampleSequencingObjectJoin createSequencingObjectInSample = sequencingObjectService
-				.createSequencingObjectInSample(sequenceFilePair, sample);
+		try {
+			// add the files and join
+			SampleSequencingObjectJoin createSequencingObjectInSample = sequencingObjectService.createSequencingObjectInSample(
+					sequenceFilePair, sample);
+			// clean up the temporary files.
+			Files.deleteIfExists(target1);
+			Files.deleteIfExists(temp1);
+			Files.deleteIfExists(target2);
+			Files.deleteIfExists(temp2);
+			logger.trace("Deleted temp files");
 
-		// clean up the temporary files.
-		Files.deleteIfExists(target1);
-		Files.deleteIfExists(temp1);
-		Files.deleteIfExists(target2);
-		Files.deleteIfExists(temp2);
-		logger.trace("Deleted temp files");
+			SequencingObject sequencingObject = createSequencingObjectInSample.getObject();
 
-		SequencingObject sequencingObject = createSequencingObjectInSample.getObject();
-
-		sequencingObject = addSequencingObjectLinks(sequencingObject, sampleId);
-
-		sequencingObject.add(linkTo(methodOn(RESTSampleSequenceFilesController.class).getSampleSequenceFiles(sampleId))
-				.withRel(REL_SAMPLE_SEQUENCE_FILES));
+			sequencingObject = addSequencingObjectLinks(sequencingObject, sampleId);
 
-		// add location header
-		response.addHeader(HttpHeaders.LOCATION, sequencingObject.getLink("self").getHref());
+			sequencingObject.add(
+					linkTo(methodOn(RESTSampleSequenceFilesController.class).getSampleSequenceFiles(sampleId)).withRel(
+							REL_SAMPLE_SEQUENCE_FILES));
+
+			// add location header
+			response.addHeader(HttpHeaders.LOCATION, sequencingObject.getLink("self")
+					.getHref());
+
+			// set the response status.
+			response.setStatus(HttpStatus.CREATED.value());
+			modelMap.addAttribute(RESTGenericController.RESOURCE_NAME, sequencingObject);
+		} catch (IllegalArgumentException e) {
+			response.setStatus(HttpStatus.BAD_REQUEST.value());
```

Comment:
Throw a logging statement here with the exception please so it doesn't just get swallowed.  `debug` level is probably fine.

Reply:
Fixed in 1d9c20b27fa35533f4334f521558c3e28baef6ae